src/hooks/useWorkspaceFileSystemSynchronization.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+import { fileSystemManager } from '../managers/FileSystemManager';
+import { useAppDispatch } from '../state/store';
+import { setWorkspaces } from '../state/workspaces/slice';
+
+export function useWorkspaceFileSystemSynchronization() {
```

Comment:
This hook just sets up the FileSystem listener for workspaces and also clears it when unmounted.